Source/Projections/Builder/TypedClassProjectionMethod.cs

```diff
@@ -0,0 +1,100 @@
+        /// </summary>
+        /// <param name="method">The <see cref="TaskResultProjectionMethodSignature{TProjection, TEvent}"/> method to invoke.</param>
+        /// <param name="keySelector">The <see cref="Projections.KeySelector" />.</param>
+        public TypedClassProjectionMethod(TaskResultProjectionMethodSignature<TProjection, TEvent> method, KeySelector keySelector)
+        {
+            _method = method;
+            KeySelector = keySelector;
+        }
+
+        /// <summary>
+        /// Initializes a new instance of the <see cref="TypedClassProjectionMethod{TProjection, TEvent}"/> class.
+        /// </summary>
+        /// <param name="method">The <see cref="SyncProjectionMethodSignature{TProjection, TEvent}" /> method to invoke.</param>
+        /// <param name="keySelector">The <see cref="Projections.KeySelector" />.</param>
+        public TypedClassProjectionMethod(
+            SyncProjectionMethodSignature<TProjection, TEvent> method,
+            KeySelector keySelector)
+            : this(
+                (TProjection instanceAndReadModel, TEvent @event, ProjectionContext context) =>
+                {
+                    method(instanceAndReadModel, @event, context);
+                    return ProjectionResultType.Replace;
+                },
+                keySelector)
+        {
+        }
+
+        /// <summary>
+        /// Initializes a new instance of the <see cref="TypedClassProjectionMethod{TProjection, TEvent}"/> class.
+        /// </summary>
+        /// <param name="method">The <see cref="SyncResultProjectionMethodSignature{TProjection, TEvent}" /> method to invoke.</param>
+        /// <param name="keySelector">The <see cref="Projections.KeySelector" />.</param>
+        public TypedClassProjectionMethod(
+            SyncResultProjectionMethodSignature<TProjection, TEvent> method,
+            KeySelector keySelector)
+            : this(
+                (TProjection instanceAndReadModel, TEvent @event, ProjectionContext context)
+                    => Task.FromResult(method(instanceAndReadModel, @event, context)),
+                keySelector)
+        {
+        }
+
+        /// <inheritdoc/>
+        public KeySelector KeySelector { get; }
+
+        /// <inheritdoc/>
+        public EventType GetEventType(IEventTypes eventTypes)
+            => eventTypes.GetFor(typeof(TEvent));
+
+        /// <inheritdoc/>
+        public async Task<Try<ProjectionResult<TProjection>>> TryOn(TProjection readModel, object @event, ProjectionContext context)
+        {
+            if (@event is TEvent typedEvent)
+            {
+                var resultType = await _method(readModel, typedEvent, context).TryTask().ConfigureAwait(false);
+                if (resultType.Exception != null) return resultType.Exception;
+                var result = new Try<ProjectionResult<TProjection>>(readModel);
+                return result;
```

Comment:
Same here as previous comment - seems to be impossible to perform a delete in a projection class.